MapboxVision/src/test/java/com/mapbox/vision/telemetry/TotalBytesCounterTest.kt

```diff
@@ -0,0 +1,291 @@
+                actualValue = totalBytesCounter.fitInLimit(DEFAULT_SESSION_SIZE_MB * 1024 * 1024L)
+            }
+
+            Then("It should be <true>") {
+                assertEquals(true, actualValue)
+            }
+
+            When("Get fitInLimit for ${DEFAULT_SESSION_SIZE_MB}MB and 1 byte") {
+                actualValue = totalBytesCounter.fitInLimit(DEFAULT_SESSION_SIZE_MB * 1024 * 1024L + 1)
+            }
+
+            Then("It should be <false>") {
+                assertEquals(false, actualValue)
+            }
+        }
+
+        Scenario("Check default session length") {
+            lateinit var totalBytesCounter: TotalBytesCounter
+            val mockedTime = mockk<Time>()
+            every { mockedTime.millis() }.returns(0)
+
+            Given("TotalBytesCounter with default params") {
+                totalBytesCounter = TotalBytesCounter.Impl(time = mockedTime)
+            }
+
+            var actualValue = 0L
+
+            When("Get millisToNextSession") {
+                actualValue = totalBytesCounter.millisToNextSession()
+            }
+
+            Then("It should be <1 hour>") {
+                assertEquals(TimeUnit.HOURS.toMillis(1), actualValue)
+            }
+        }
+
+        Scenario("Check fitInLimit") {
+
+            lateinit var totalBytesCounter10Min10kBytes: TotalBytesCounter
+            val mockedTime = mockk<Time>()
+
+            val testCases = mapOf(
+                Long.MIN_VALUE to true,
+                -1L to true,
+                -10_000L to true,
+                -10_001L to true,
+                -15_000L to true,
+                0L to true,
+                5_000L to true,
+                10_000L to true,
+
+                10_001L to false,
+                15_000L to false,
+                20_000L to false,
+                Long.MAX_VALUE to false
+            )
+
+            Given("TotalBytesCounter with 10 minutes, 10 KBytes params") {
+                totalBytesCounter10Min10kBytes = TotalBytesCounter.Impl(
+                    sessionLengthMillis = TEST_SESSION_LENGTH_MS,
+                    sessionMaxBytes = TEST_SESSION_MAX_BYTES,
+                    time = mockedTime
+                )
+            }
+
+            testCases.forEach { (bytes, expectedValue) ->
+                var actualValue = false
+
+                When("Get FitInLimit for $bytes bytes") {
+                    actualValue = totalBytesCounter10Min10kBytes.fitInLimit(bytes)
+                }
+
+                Then("It should be <$expectedValue>") {
+                    assertEquals(expectedValue, actualValue)
+                }
+            }
+        }
+
+        Scenario("Check millisToNextSession") {
+
+            lateinit var totalBytesCounter10Min10kBytes: TotalBytesCounter
+            val mockedTime = mockk<Time>()
+
+            val testCases = mapOf(
+                TimeUnit.MINUTES.toMillis(0) to TimeUnit.MINUTES.toMillis(10),
```

Comment:
Would you like to simplify it to hold seconds, and just map it after? Or even before usage